### ISSUE TITLE
"Invalid IdP response/credential" error using signInWithPopup with Google, Facebook, or Twitter providers and Firebase Emulators Suite

### ISSUE THREAD

**Issue by John-Shortland-MaiMarae** [firebase/firebase-tools#3502]
### [REQUIRED] Environment info

**firebase-tools:** 9.13.0

**Platform:** macOS 11.4

### [REQUIRED] Test case

My React web app can sign-in using the Google, Facebook & Twitter providers when connecting to production; however, I'm trying to use the Firebase Emulators Suite and upon signing-in using any of the providers, I get an `auth/internal-error` error.

### [REQUIRED] Steps to reproduce

1. Load Home screen, then this code is called: `firebase.auth().useEmulator(`http://localhost:9099`);`
2. Click on Google icon, then this code is called: `const result = await firebase.auth().signInWithPopup(new firebase.auth.GoogleAuthProvider());`
3. Sign-in with valid user credentials to Popup window, then this `auth/internal-error` error is returned with `message: Invalid IdP response/credential: https://my-react-app.firebaseapp.com/__/auth/handler?state=XXX&code=YYY&scope=ZZZ&authuser=2&prompt=none'`

### [REQUIRED] Expected behavior

Expected call to return with a `Promise<UserCredential>` object from Google Sign-in provider as per the documentation (https://firebase.google.com/docs/reference/js/firebase.auth.Auth#signinwithpopup).

### [REQUIRED] Actual behavior

Thrown Error object: `{
  code: "auth/internal-error",
  message: "Invalid IdP response/credential: https://my-fake-app.firebaseapp.com/__/auth/handler?state=XXX&code=YYY&scope=ZZZ&authuser=2&prompt=none"
}`

**Comment by samtstern** [comment]
Hmm `https://my-react-app.firebaseapp.com/__/auth/handler?state=XXX&code=YYY&scope=ZZZ&authuser=2&prompt=none` shouldn't be involved at all if you're using `useEmulator`. I would expect to see a `localhost` URL.

Are you definitely calling `useEmulator` before `signInWithPopup`? And in both cases are you using `firebase.auth()` and not `app.auth()` or some other non-default FirebaseAuth instance>

**Comment by John-Shortland-MaiMarae** [comment]
> Hmm `https://my-react-app.firebaseapp.com/__/auth/handler?state=XXX&code=YYY&scope=ZZZ&authuser=2&prompt=none` shouldn't be involved at all if you're using `useEmulator`. I would expect to see a `localhost` URL.

Agreed, I expected to a `localhost` URL too; also, I can confirm `http://localhost`, by default, is a registered Authorized JavaScript origins > Client ID for Web application > OAuth 2.0 Client IDs > APIs & Services > Google Cloud Platform.

> Are you definitely calling `useEmulator` before `signInWithPopup`? And in both cases are you using `firebase.auth()` and not `app.auth()` or some other non-default FirebaseAuth instance?

Yes I am. I've updated my example to clarify a bit better.

What's frustrating is, this works in production; just not using local emulators?!

Hope this helps, thank @samtstern for taking the time to review.

**Comment by samtstern** [comment]
@John-Shortland-MaiMarae let's try to simplify things as much as possible.

You have this:
```js
const firebase = firebaseApp.initializeApp(config);
...
React.useEffect(() => {
	if (process.env.NODE_ENV !== 'production') {
		firebase.auth().useEmulator('http://localhost:9099');
	}
}, []);
```

There are two complexities there:

  * `useEffect`
  * `NODE_ENV`

Instead can you try just simplifying your code to this:

```
const firebase = firebaseApp.initializeApp(config);
firebase.auth().useEmulator('http://localhost:9099');
```

And then try again. Then we'll have two possibiltiies:

  * If it works it means you have some error in your React setup, and there won't be anything we can do here.
  * If it doesn't work it means there's something going wrong in either the Firebase SDK or CLI and we'll dig in more.

**Comment by John-Shortland-MaiMarae** [comment]
Unfortunately, the suggested simplifications didn't make a difference.

Thanks again @samtstern, I sincerely appreciate your efforts.
